app/src/main/java/sdp/moneyrun/ui/game/GameLobbyActivity.java

```diff
@@ -1,156 +1,211 @@
+                        game.removePlayer(user, false);
+                    }
+                }
+
+                @Override
+                public void onCancelled(@NonNull DatabaseError error) {
+                    Log.e(TAG, error.getMessage().toString());
+                }
+            });
+        }
+    }
+
+    private void createDeleteOrLeaveButton() {
+        if (user.equals(game.getHost())) {
+            Button leaveButton = (Button) findViewById(R.id.leave_lobby_button);
+            leaveButton.setText("Delete");
+            leaveButton.setOnClickListener(getDeleteClickListener());
+        } else {
+            findViewById(R.id.leave_lobby_button).setOnClickListener(getLeaveClickListener());
+        }
+    }
+
+    private void setAllFieldsAccordingToGame() {
+        //Find all the views and assign them values
+        TextView name = (TextView) findViewById(R.id.lobby_title);
+        name.setText(game.getName());
+
+        findViewById(R.id.launch_game_button).setOnClickListener(v -> {
+            if (game.getHost().equals(user)) {
+                game.setStarted(true, false);
+                Intent intent = new Intent(getApplicationContext(), MapActivity.class);
+                intent.putExtra("player", user);
+                intent.putExtra("gameId", gameId);
+                intent.putExtra("host", true);
+                startActivity(intent);
+                finish();
+            }
+        });
+
+        //Player List is dynamic with DB
+        TextView playersMissing = (TextView) findViewById(R.id.players_missing_TextView);
+        thisGame.child(DB_PLAYERS).addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot snapshot) {
+                GenericTypeIndicator<List<Player>> t = new GenericTypeIndicator<List<Player>>() {
+                };
+                List<Player> newPlayers = snapshot.getValue(t);
+                listAdapter.clear();
+                addPlayerList(new ArrayList<Player>(newPlayers));
+                String newPlayersMissing = getString(R.string.lobby_player_missing, game.getMaxPlayerCount() - newPlayers.size());
+
+                playersMissing.setText(newPlayersMissing);
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+                Log.e(TAG, error.getMessage());
+            }
+        });
+    }
+
+
+    private View.OnClickListener getDeleteClickListener() {
+        return v -> {
+            game.setIsDeleted(true, false);
+            thisGame.child(DB_PLAYERS).addValueEventListener(new ValueEventListener() {
+                @Override
+                public void onDataChange(@NonNull DataSnapshot snapshot) {
+                    GenericTypeIndicator<List<Player>> t = new GenericTypeIndicator<List<Player>>() {};
+                    List<Player> players = snapshot.getValue(t);
+                    if (players.size() == 1) {

```

Comment:
Is this because you are waiting to be the last player to leave? And shouldn't you delete the Game as well from the DB?

Reply:
Yes, waiting to be the last, so that when we delete it other players arent listening to certain fields and then we end up getting a nullPointer.

Yes the goal was to delete it, but there is some listener somewhere which gives me a nullPointer Exception if I delete the game, I tried to find the source of the problem for a few hours with no conclusion. Making it invisible has the same effect (at least for the app user). In regards to actually deleting it, I will make a small task next sprint to try and solve the problem. but it feels unfair for this detail to hold back the entire PR as all other functionalities work.

Reply:
I don't think this is a good practice, to be honest, I think you should find the bug before merging the PR.

Reply:
Ok, I changed it so that now it actually deletes it.